claw/ContentView.swift

```diff
@@ -131,6 +131,7 @@ struct ContentView: View {
             }).environmentObject(settings).environment(\.managedObjectContext, viewContext)
         }.environment(\.didReselect, didReselect.eraseToAnyPublisher()).accentColor(settings.accentColor).font(Font(.body, sizeModifier: CGFloat(settings.textSizeModifier))).onOpenURL(perform: { url in
             let _ = print(url)
+            UIApplication.shared.windows.first?.rootViewController?.presentedViewController?.dismiss(animated: true, completion: nil)
```

Comment:
Can you add a check that the there is a sheet to dismiss? If self.observableSheet.sheet is nil? also only dismiss if you are about to present a new sheet. Can you just set self.observableSheet.sheet to nil to dismiss? If not set self.observableSheet.sheet in the completion block?

Reply:
Sadly I believe it is impossible in SwiftUI to have unlimited sheets (just keep adding one on top)

Reply:
there is it's possible unlimited sheets but problem is when you click share then we are not able to identify the presented sheet that's is the issue so when we presenting new sheet first I will dismiss current presented all sheet then open new sheet

Reply:
Can't we just keep track of the top then for the share sheet?

Reply:
Let's create a ticket for unlimited sheets and just fix this bug for now

Reply:
ok sir

Reply:
The original comment still applies.